RecFCCeeCalorimeter/src/components/CalibrateBenchmarkMethod.cpp

```diff
@@ -0,0 +1,279 @@
+#include "CalibrateBenchmarkMethod.h"
+
+// Key4HEP
+#include "k4Interface/IGeoSvc.h"
+
+// datamodel
+#include "edm4hep/CalorimeterHit.h"
+#include "edm4hep/CalorimeterHitCollection.h"
+
+// DD4hep
+#include "DD4hep/Detector.h"
+#include "DD4hep/Readout.h"
+
+#include "GaudiKernel/ITHistSvc.h"
+
+#include "TH1F.h"
+#include "TVector3.h"
+#include "TMath.h"
+#include "Math/Minimizer.h"
+#include "Math/Factory.h"
+#include "Math/Functor.h"
+#include "TRandom2.h"
+#include "TError.h"
+
+std::vector<double> vecEgenerated;
+std::vector<double> vecEinECal_total;
+std::vector<double> vecEinHCal_total;
+std::vector<double> vecEinHCal_firstLayer;
+std::vector<double> vecEinECal_lastLayer;
+
+DECLARE_COMPONENT(CalibrateBenchmarkMethod)
+
+
+CalibrateBenchmarkMethod::CalibrateBenchmarkMethod(const std::string& aName, ISvcLocator* aSvcLoc)
+    : GaudiAlgorithm(aName, aSvcLoc), 
+    m_geoSvc("GeoSvc", aName), 
+    m_histSvc("THistSvc", aName),
+    h_parameter(nullptr),
+    m_totalEnergyECal(nullptr),
+    m_totalEnergyHCal(nullptr),
+    m_totalEnergyBoth(nullptr)
+    {
+      declareProperty("ecalBarrelCells", m_ecalBarrelCells, "");
+      declareProperty("hcalBarrelCells", m_hcalBarrelCells, "");
+    }
+
+
+CalibrateBenchmarkMethod::~CalibrateBenchmarkMethod() {}
+
+
+StatusCode CalibrateBenchmarkMethod::initialize() {
+  if (GaudiAlgorithm::initialize().isFailure()) return StatusCode::FAILURE;
+
+  // Check geometry service
+  if (!m_geoSvc) {
+    error() << "Unable to locate Geometry Service! "
+            << "Make sure you have GeoSvc and SimSvc in the right order in the configuration." << endmsg;
+    return StatusCode::FAILURE;
+  }
+
```

Comment:
One can check also for the histogram service

Reply:
fixed